### PR TITLE
[JBPM-10053]-UserTaskInstanceWithCustomVarsQueryMapper not returning "COLUMN_TASK_SLA_DUE_DATE"

### PR DESCRIPTION
**Pull request by VaniHaripriya**
Jira- https://issues.redhat.com/browse/JBPM-10053

Updated UserTaskInstanceWithCustomVarsQueryMapper with the missing column name- COLUMN_TASK_SLA_DUE_DATE

### CODE REVIEW

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Compilation failure, could you check?

```
jbpm-kie-services: Compilation failure
19:13:10  [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-main.pr/bc/kiegroup_jbpm/jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/UserTaskInstanceDesc.java:[54,9] reference to UserTaskInstanceDesc is ambiguous
19:13:10  [ERROR]   both constructor UserTaskInstanceDesc(java.lang.Long,java.lang.String,java.util.Date,java.lang.String,java.lang.String,java.lang.Integer,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.Long,java.util.Date,java.util.Date,java.lang.Long) in org.jbpm.kie.services.impl.model.UserTaskInstanceDesc and constructor UserTaskInstanceDesc(java.lang.Long,java.lang.String,java.util.Date,java.lang.String,java.lang.String,java.lang.Integer,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.Long,java.util.Date,java.util.Date,java.util.Date) in org.jbpm.kie.services.impl.model.UserTaskInstanceDesc match
```

**Comment by VaniHaripriya** [comment]
Jenkins retest this

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithCustomVarsQueryMapper.java [hunk 1]
```diff
@@ -96,7 +96,8 @@ protected UserTaskInstanceWithVarsDesc buildInstance(DataSet dataSetResult, int
                 getColumnStringValue(dataSetResult, COLUMN_TASK_PROCESSID, index),//processId,
                 getColumnLongValue(dataSetResult, COLUMN_TASK_PROCESSINSTANCEID, index),//processInstanceId,
                 getColumnDateValue(dataSetResult, COLUMN_CREATEDON, index),//createdOn,
-                getColumnDateValue(dataSetResult, COLUMN_DUEDATE, index)//dueDate
+                getColumnDateValue(dataSetResult, COLUMN_DUEDATE, index),//dueDate
+                getColumnDateValue(dataSetResult, COLUMN_TASK_SLA_DUE_DATE, index)//slaDueDate
```
Comment (gmunozfe) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithCustomVarsQueryMapper.java:100]: Apart from that column, shouldn't we add the rest of missing ones?

```
getColumnStringValue(dataSetResult, COLUMN_SUBJECT, index),
getColumnStringValue(dataSetResult, COLUMN_CORRELATIONKEY, index), 
getColumnIntValue(dataSetResult, COLUMN_PROCESSTYPE, index),
getColumnIntValue(dataSetResult, COLUMN_TASK_SLA_COMPLIANCE, index)
```

@VaniHaripriya @fjtirado wdyt?

**Review by gmunozfe — COMMENTED** [review]
One question to add other columns and a request for having a test on this

**Comment by VaniHaripriya** [comment]
Jenkins retest this

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/TaskVariablesQueryServiceTest.java [hunk 2]
```diff
@@ -579,7 +578,95 @@ public void testTaskVariableQueryOnBigTaskSetAsJPA() throws Exception {
         
 
     }
-    
+
+
+    @Test
+    public void testUserTaskInstanceWithCustomVarsQueryMapperResult() throws Exception {
```
Comment (gmunozfe) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/TaskVariablesQueryServiceTest.java:584]: This test is part of an "ignore" annotated class, so won't be executed in CI. Anyway, you can keep it for reference

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithCustomVarsQueryMapper.java [hunk 3]
```diff
@@ -96,7 +96,12 @@ protected UserTaskInstanceWithVarsDesc buildInstance(DataSet dataSetResult, int
                 getColumnStringValue(dataSetResult, COLUMN_TASK_PROCESSID, index),//processId,
                 getColumnLongValue(dataSetResult, COLUMN_TASK_PROCESSINSTANCEID, index),//processInstanceId,
                 getColumnDateValue(dataSetResult, COLUMN_CREATEDON, index),//createdOn,
-                getColumnDateValue(dataSetResult, COLUMN_DUEDATE, index)//dueDate
+                getColumnDateValue(dataSetResult, COLUMN_DUEDATE, index),//dueDate
+                getColumnStringValue(dataSetResult, COLUMN_SUBJECT, index),//subject
+                getColumnStringValue(dataSetResult, COLUMN_CORRELATIONKEY, index),//correlationKey
+                getColumnIntValue(dataSetResult, COLUMN_PROCESSTYPE, index),//processType
+                getColumnDateValue(dataSetResult, COLUMN_TASK_SLA_DUE_DATE, index),//slaDueDate
+                getColumnIntValue(dataSetResult, COLUMN_TASK_SLA_COMPLIANCE, index)//slaCompliance
```
Comment (gmunozfe) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithCustomVarsQueryMapper.java:104]: Same here for COLUMN_WORKITEMID and COLUMN_FORM_NAME

**Review by gmunozfe — CHANGES_REQUESTED** [review]
include also COLUMN_WORKITEMID and COLUMN_FORM_NAME

**Review by gmunozfe — APPROVED** [review]
Looks good to me, well done @VaniHaripriya !

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/UserTaskInstanceDesc.java [hunk 4]
```diff
@@ -126,7 +126,7 @@ public UserTaskInstanceDesc(Long taskId, String status, String actualOwner,
         this.deploymentId = deploymentId;
         this.dueDate = dueDate;
     }
-    
+
```
Comment (fjtirado) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/UserTaskInstanceDesc.java:129]: I think this files does not need to be modified, you can checkout previous version from history and commit

**Review by fjtirado — APPROVED** [review]

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
@VaniHaripriya could you backport it to 7.67.x as well?

**Pull request merged**